### PR TITLE
Fixing build related issues for enabling IPv6 in libcurl

### PR DESCRIPTION
**Pull request by aravindhank11**
Signed-off-by: Aravindhan K <aravindhank11@gmail.com>

### CODE REVIEW

**Comment by aravindhank11** [comment]
The build code for TENS is dated and has quite a few installations that is not needed.
Also when TLS 1.3 support was added we had custom compiled libcurl against libssl 1.1, and during this process we had missed to enabled ipv6 support for curl causing the breakage.

File: te/tedp_docker/setup.sh [hunk 1]
```diff
@@ -29,50 +29,60 @@
 # OF SUCH DAMAGE
 #**********************************************************************************************
 
-#!/bin/sh
-apt update  # To get the latest package lists
+#!/bin/bash
 cd /tmp/
+sed -i '/deb-src/s/^# //' /etc/apt/sources.list && apt update
```
Comment (aravindhank11) [inline te/tedp_docker/setup.sh:34]: we wanted to install pre-depdencies of curl using `apt build-dep curl -y`. This needs deb-src to be uncommented in sources.list

File: te/tedp_docker/setup.sh [hunk 2]
```diff
@@ -29,50 +29,60 @@
...
 apt install automake libtool m4 -y
 apt install libjson0 libjson0-dev -y
 apt build-dep curl -y
 apt install wget -y
-apt install build-essential nghttp2 libnghttp2-dev libssl-dev -y
+apt install build-essential nghttp2 libnghttp2-dev -y
```
Comment (aravindhank11) [inline te/tedp_docker/setup.sh:39]: We are using custom compiled libssl1.1 and thus `libssl-dev` is moot

File: te/tedp_docker/setup.sh [hunk 3]
```diff
@@ -29,50 +29,60 @@
...
+
+###################### Download all first ######################
```
Comment (aravindhank11) [inline te/tedp_docker/setup.sh:41]: Do the downloads followed by compilation. `wget` is a temporary package that depends on libssl1.0. So post getting the source code, libssl can be uninstalled

File: te/tedp_docker/setup.sh [hunk 4]
```diff
@@ -29,50 +29,60 @@
...
+rm -rf openssl*
+
+#LIBCURL
 cd curl-7.67.0
-#./configure --with-nghttp2 --prefix=/usr/local --with-ssl=/usr/local/ssl
 #For openssl 1.1.1 support (TLSv1.3)
-./configure --with-nghttp2 --prefix=/usr/local --with-default-ssl-backend=openssl
-make -j8
-make install
+./configure --with-nghttp2 --prefix=/usr/local --with-default-ssl-backend=openssl --enable-ipv6
```
Comment (aravindhank11) [inline te/tedp_docker/setup.sh:75]: enabling ipv6 support

File: te/tedp_docker/tedp_bin_builder.dockerfile [hunk 5]
```diff
@@ -41,35 +41,32 @@ ARG lib64_path=/lib/x86_64-linux-gnu
 # basic library and pkg install
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 RUN apt update && \
-    apt install -y libssl-dev -y libffi-dev  && \
```
Comment (aravindhank11) [inline te/tedp_docker/tedp_bin_builder.dockerfile:44]: installing and uninstalling libssl-dev -- not needed

**Comment by priyadarshitathagat** [comment]
Thanks AK!

**Pull request merged**